packages/storage/amplify_storage_s3_dart/lib/src/model/storage_s3_item.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_storage_s3_dart/amplify_storage_s3_dart.dart';
+import 'package:amplify_storage_s3_dart/src/sdk/s3.dart';
+import 'package:meta/meta.dart';
+
+/// {@template storage.amplify_storage_s3.storage_s3_item}
+/// This defines an object in a S3 bucket.
+/// {@endtemplate}
+class StorageS3Item extends StorageItem {
+  /// {@macro storage.amplify_storage_s3.storage_s3_item}
+  StorageS3Item({
+    required super.key,
+    super.size,
+    super.lastModified,
+    super.eTag,
+    this.metadata = const <String, String>{},
+    this.versionId,
+  });
+
+  /// Creates a [StorageS3Item] from [S3Object] provided by smithy.
+  /// This named constructor should be used internally only.
+  @internal
+  factory StorageS3Item.fromS3Object(
+    S3Object object, {
+    required String prefixToDrop,
+  }) {
+    final key = object.key;
+
+    // In S3 plugin, key is required property presenting an object
+    if (key == null) {
+      throw StorageS3Exception.unknownException();
+    }
+
+    final keyDroppedPrefix = key.replaceRange(0, prefixToDrop.length, '');
+
+    return StorageS3Item(
+      key: keyDroppedPrefix,
+      size: object.size?.toInt(),
+      lastModified: object.lastModified,
+      eTag: object.eTag,
+    );
+  }
+
+  /// Metadata specified when the object was uploaded.
+  final Map<String, String> metadata;
+
+  /// Object `versionId`, may be available when S3 bucket versioning is enabled.
+  final String? versionId;
+}
+
+/// A list of [StorageS3Item].
+typedef StorageS3Items = StorageItems<StorageS3Item>;
```

Comment:
Is `StorageItems` a type we already have in stable?

What is the advantage of creating this typedef? I personally like the explicit `List<StorageS3Item>`. When I see `StorageS3Items` or `StorageItems` I have to remind myself what that is.

Reply:
If you do feel that there is an advantage to keeping this, I would suggest renaming it `StorageItemList`/`StorageS3ItemList`. That would make it clear that it is a list.

Reply:
Yea this alias is not really public facing... can remove. 👍